com.vrcfury.vrcfury/Editor/VF/Service/ScalePropertyCompensationService.cs

```diff
@@ -0,0 +1,120 @@
+                .Where(path => animatedPaths.Contains(path))
+                .Where(path => path != "") // VRChat ignores animations of the root scale now, so we need to as well
+                .ToList();
+
+            var pathToParam = new Dictionary<string, VFAFloat>();
+            var pathNumber = 0;
+            foreach (var path in animatedParentPaths) {
+                pathNumber++;
+                var param = manager.GetFx().NewFloat("shaderScale_" + objectNumber + "_" + pathNumber, def: manager.AvatarObject.transform.Find(path).localScale.z);
+                pathToParam[path] = param;
+                Debug.Log(path + " " + param.Name());
+            }
+            foreach (var clip in manager.GetFx().GetClips()) {
+                foreach (var binding in clip.GetFloatBindings()) {
+                    if (!IsScaleBinding(binding)) continue;
+                    if (!pathToParam.TryGetValue(binding.path, out var param)) continue;
+                    var newBinding = new EditorCurveBinding();
+                    newBinding.type = typeof(Animator);
+                    newBinding.path = "";
+                    newBinding.propertyName = param.Name();
+                    clip.SetFloatCurve(newBinding, clip.GetFloatCurve(binding));
+                }
+            }
+
+            float handledScale = 1;
+            foreach (var path in animatedParentPaths) {
+                handledScale *= manager.AvatarObject.transform.Find(path).localScale.z;
+            }
+
+            if (directTree == null) {
+                Debug.Log("Creating direct layer");
+                var layer = manager.GetFx().NewLayer("shaderScale");
+                var state = layer.NewState("Scale");
+                directTree = manager.GetFx().NewBlendTree("shaderScale");
+                directTree.blendType = BlendTreeType.Direct;
+                state.WithAnimation(directTree);
+
+                zeroClip = manager.GetFx().NewClip("zeroScale");
+                var one = manager.GetFx().One();
+                directTree.AddDirectChild(one.Name(), zeroClip);
+            }
+
+            var scaleClip = manager.GetFx().NewClip("tpsScale_" + objectNumber);
+            foreach (var prop in properties) {
+                objectNumber++;
+                Debug.Log("Processing " + prop.ObjectPath);
+
+                if (prop.InitialValue is float f) {
+                    var lengthOffset = f / handledScale;
+                    scaleClip.SetCurve(prop.ObjectPath, prop.ComponentType, prop.PropertyName, ClipBuilderService.OneFrame(lengthOffset));
+                    zeroClip.SetCurve(prop.ObjectPath, prop.ComponentType, prop.PropertyName, ClipBuilderService.OneFrame(0));
+                } else if (prop.InitialValue is Vector4 vec) {
+                    var scaleOffset = vec.z / handledScale;
+                    scaleClip.SetCurve(prop.ObjectPath, prop.ComponentType, prop.PropertyName, ClipBuilderService.OneFrame(scaleOffset));
```

Comment:
This mysteriously lost the .x, .y, and .z suffixes.

Reply:
woops, didn't notice that